app/controllers/user.rb

```diff
@@ -0,0 +1,43 @@
+get '/sign_up' do
+  erb :'user/sign_up'
+end
+
+post '/sign_up' do
+  @user = User.new(params[:user])
+  @user_exists = false
+
+  if User.find_by(username: @user.username)
+    "A user already exists with this username, please choose another"
+    @user_exists = true
+    return erb :'user/sign_up'
+  end
+
+  if @user.save
+    return "go to profile"
+  end
+
+  "something went wrong"
+
+end
+
+get '/login' do
+  erb :'user/login'
+end
+
+
+
+post '/login' do
+  @user = User.find_by(username: params[:user][:username])
+  @user_not_found = false
+
+  if @user.nil?
```

Comment:
I think you can combine this conditional on line 33  and on line 38 into one if/else. Something like if @user && @user.password == params[key], do this, else return error 'username or email not found' or redirect wherever you'd like. This is authentication but for authorization you will need to utilize sessions, so research that and/or look at anonymous blog-2.